src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -50,11 +51,21 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(LibraryIde
 
                 var projectFullPath = Path.Combine(projectDirectory, $"{projectName}.proj");
 
-                // The package spec details what packages to restore
-                var packageSpec = new PackageSpec(TargetFrameworks.Select(i => new TargetFrameworkInformation
+                // Iterate through TargetFrameworks to generate Lists required for packageSpec
+                var tfi = new List<TargetFrameworkInformation>();
+                var otf = new List<string>();
```

Comment:
```suggestion
var otf = new List<string>(tfi.Capacity);
```

I'd check that it works first. 

Or you can specify `1` instead.